core/dbt/context/providers.py

```diff
@@ -174,30 +175,34 @@ def dispatch(
             raise MacroDispatchArgError(macro_name)
 
         search_packages = self._get_search_packages(macro_namespace)
-
-        attempts = []
+        macro = None
+        potential_macros: List[Tuple[Optional[str], str]] = []
+        failed_macros: List[Tuple[Optional[str], str]] = []
 
         for package_name in search_packages:
-            for prefix in self._get_adapter_macro_prefixes():
-                search_name = f"{prefix}__{macro_name}"
-                try:
-                    # this uses the namespace from the context
-                    macro = self._namespace.get_from_package(package_name, search_name)
-                except CompilationError:
-                    # Only raise CompilationError if macro is not found in
-                    # any package
-                    macro = None
-
-                if package_name is None:
-                    attempts.append(search_name)
-                else:
-                    attempts.append(f"{package_name}.{search_name}")
-
-                if macro is not None:
+            if macro_name.startswith("materialization_"):
```

Comment:
I think we made a fix recently for scenarios where macros that started with "materialization_" were incorrectly being picked up as materializations. For example, this was being flagged as a materialization instead of a macro:
```sql
{% macro materialization_setup() %}
    ...
{% endmacro %}
```
Have you verified this doesn't reintroduce that issue?